crates/flux-refineck/src/checker.rs

```diff
@@ -1531,11 +1675,11 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
         &self.inherited.ghost_stmts[&self.def_id]
     }
 
-    fn refine_default(&self, ty: &ty::Ty) -> QueryResult<Ty> {
+    fn refine_default<T: Refine>(&self, ty: &T) -> QueryResult<<T as Refine>::Output> {
```

Comment:
I think you don't need the `as Refine`

```suggestion
    fn refine_default<T: Refine>(&self, ty: &T) -> QueryResult<T::Output> {
```